### ISSUE TITLE
Agent doesn't learn a simple game of goal scoring

### ISSUE THREAD

**Issue by roboserg** [Unity-Technologies/ml-agents#2023]
So the game looks like [this ](https://puu.sh/Dq192/d5666195d1.png) where a checkerboard agent has to score the orange ball into the blue net. As you can see in this [video](https://puu.sh/Dq14h/c526d9faad.mp4), the agent gets stuck after **1 million** episodes.

Reward - only when the ball touches the net.

Actions - two cont. actions of moving along each axis (X and Z).

Observations: absolute X and Z coordinates of the position of every object + agent velocity, 8x observations in total.

Tensorboard - https://puu.sh/Dq1HR/63091d264d.png

Hyperparameters - https://puu.sh/Dq1Jx/c178d23d24.png

1. Should I maybe use relative coordinates, the difference between an agent and the ball, etc? 
2. Any ideas on how to solve this problem? 
3. How do I choose the hyperparameters? RollerAgent from the tutorial used   batch_size: 10  and  buffer_size: 100 and learned fine. Other environments use number 10's to 100's times bigger.
4. Maybe the reward is too sparse? Maybe I should add a reward continuously as a function of the distance between the ball and the net or smth similar?  

ps. RollerAgent from the tutorial works fine.

**Comment by ervteng** [comment]
Hi @roboserg, your scenario is very similar to the PushBlock example in the Examples, I'd take a look at that one to set your observations and hyper parameters. 

For more complex tasks with sparse reward such as this one, you may want to increase the buffer size from 100 to ~ 512, and increase the batch size as well, that way during an update step, at least one instance of solving the problem exists in the buffer. 

A dense reward will learn faster, but it should be able to solve this task eventually even with the sparse reward.

**Comment by roboserg** [comment]
@ervteng thanks. Yea, PushBlock is similar, but it uses RayCasts, I specifically don't want to use it, since coordinates of positions should provide better information. The question is whether absolute or relative positions are better or both are equally good. 

As you've seen, even after 1 mio epochs it was getting stuck. That's weird since the problem isn't that much harder then RollerBall from the tutorial, which learns in about 30k epochs or less.

**Comment by roboserg** [comment]
Ok, smth. broke on my end and I can't find the issue. So I had the RollerBall tutorial example working perfectly. Since I have issues with this soccer agent, I opened RollelBall and retrained it. Something is wrong with my training, see video - https://puu.sh/Dq6kA/2b6a21b86f.mp4

The red agent was trained when I first made the tutorial, checkerboard trained now. I used the same scene, same everything. Hyperparameters I took default ones ( I tried 10 and 100 for batch and buffer also). Something is not right here, the agent doesn't learn in both scenes now.

**Comment by ervteng** [comment]
Hi @roboserg, hmm - it's hard to say what's going on, other than it's not learning :P 

Can you post your hyper parameters and the mlagents-learn command you used to train the second agent?

**Comment by roboserg** [comment]
So I redid the RollerBall tutorial from scratch - everything works. Now I came back to the scene I have the problem with, trained it today again - it doesn't learn at all, see video - https://puu.sh/Dqv3n/35a2deb33a.mp4

Hyperparameter  - https://puu.sh/Dqvcc/64f4cffb92.png
cmd command - `mlagents-learn config/config.yaml --train --run-id=RollerBall_v3-10`

Then I took the default hyperparameters, changed only batch and buffer size to 10 and 100 and magic - it worked. Hyperpara - https://puu.sh/DqvpK/6a43197ccc.png Video - https://puu.sh/DqvsY/2c006ea79e.mp4

Now I am unsure why the first set of hyperpara didnt work?
